noir-projects/noir-contracts/contracts/lending_contract/src/position.nr

```diff
@@ -1,21 +1,35 @@
 use dep::aztec::protocol_types::traits::{Deserialize, Serialize};
 
 pub struct Position {
-    collateral: Field,
-    static_debt: Field,
-    debt: Field,
+    collateral: U128,
+    static_debt: U128,
+    debt: U128,
 }
 
-global POSITION_SERIALIZED_LEN: u32 = 3;
+global POSITION_SERIALIZED_LEN: u32 = 6;
 
 impl Serialize<POSITION_SERIALIZED_LEN> for Position {
     fn serialize(position: Position) -> [Field; POSITION_SERIALIZED_LEN] {
-        [position.collateral.to_field(), position.static_debt.to_field(), position.debt.to_field()]
+        let serialized_collateral = position.collateral.serialize();
```

Comment:
Since it's now using the canonical serde, can't we derive this?

Reply:
Good idea. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/11142/commits/8ea61529548688999967b443de5b43739963b45e